sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -105,4 +123,23 @@ private static Duration getEffectiveBackoff(Duration backoff, Duration remaining
 
         return backoff;
     }
+
+    private boolean shouldRetryLocally(CosmosSessionRetryOptions sessionRetryOptions, int retryCountForRegion) {
+
+        if (sessionRetryOptions == null) {
+            return true;
+        }
+
+        CosmosRegionSwitchHint regionSwitchHint = ImplementationBridgeHelpers
+            .CosmosSessionRetryOptionsHelper
+            .getCosmosSessionRetryOptionsAccessor()
+            .getRegionSwitchHint(sessionRetryOptions);
+
+        if (regionSwitchHint == null || regionSwitchHint == CosmosRegionSwitchHint.LOCAL_REGION_PREFERRED) {
+            return true;
+        }
+
+        return !(regionSwitchHint == CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED
+            && retryCountForRegion == this.maxRetryAttemptsInCurrentRegion.get());
```

Comment:
do we need to increase the retryCount before this checking?

Reply:
Fixed in next iteration.